### DISCUSSION TITLE
Release 5.0.2

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1099] [Announcement]
Release 5.0.2
===

This release is primarily aimed at getting the UI "`Suit: ...`" line working properly.

* The "`Suit: ...`" UI line should now function as best it can given the available data from the game.  It should not appear if you have launched the Horizons version of the game, even if your account has Odyssey enabled.  You might see "`<Unknown>`" as the text when this application does not yet have the required data.

* Changed the less than obvious "`unable to get endpoint: /profile`" error message to "`Frontier CAPI query failure: /profile`", and similarly for the other CAPI endpoints we attempt to access.  This new form is potentially translated, but translators need time to do that.

  In addition the old message "`Received error {r.status_code} from server`" has been changed to "`Frontier CAPI server error: {r.status_code}`" and is potentially translated.

* The filenames used for 'Market data in CSV format file' will now be sane, and as they were before 5.0.0.

* Linux: 'Shipyard provider' will no longer default to showing 'False' if no specific provider has been selected.

Plugin Developers
---

* Extra `Flagse` values added in the live release of Odyssey have been added to `edmc_data.py`.

* Odyssey 'BackPack' values should now track better, but might still not be perfect due to Journal bugs/shortcomings.

* `state` passed to `journal_entry()` now has a `BackpackJSON` (note the case) member which is a copy of the data from the `Backpack.json` (yes, that's currently the correct case) file that is written when there's a `BackPack` (guess what, yes, that is currently the correct case) event written to the Journal.

* `state['Credits']` tracking is almost certainly not perfect.  We're accounting for the credits component of `SuitUpgrade` now, but there might be other such we've yet accounted for.

* `state['Suits']` and associated other keys should now be tracking from Journal events, where possible, as well as CAPI data.

* There is a section in PLUGINS.md about how to package an extra Python module with your plugin.  Note the new caveat in [PLUGINS.md:Avoiding-pitfalls](./PLUGINS.md#avoiding-potential-pitfalls) about the name of your plugin's directory.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.0.2'>Release 5.0.2</a>.</em>

**Comment by srseibs** [comment]
I am a long time user having a problem updating to 5.0.2. When launching, it asks to update to 5.0.2 (I think I was using 5.0.1, but I use this frequently and accept all updates). The installer ran silently but did **NOT** restart the ED Market Connector like I think is normal, so I launched EDMC manually and go on thinking the install went well.

Today I noticed though the it keeps asking me to update to 5.0.2 again, and looking more closely I see that the update never completed (looking in the running app, help, about does not show 5.0.2 as expected).

So I tried to Add/Remove but it fails to find a "EDMarketConnector_win_5.0.0.msi'

I removed all the ED Market Connector files and directories that I can locate. Add/Remove is very confused now thinking the program still exists and I am unable to execute the Uninstall.

I loaded the MSI for 5.0.2 and tried to run it manually but it too fails looking for the same file.

How do I get 5.0.2 to freshly install at this point?

**Reply by Athanasius** [reply]
> Today I noticed though the it keeps asking me to update to 5.0.2 again, and looking more closely I see that the update never completed (looking in the running app, help, about does not show 5.0.2 as expected).
> 
> So I tried to Add/Remove but it fails to find a "EDMarketConnector_win_5.0.0.msi'

This implies you were on 5.0.0 rather than 5.0.1, prior to the upgrade to 5.0.2, or things have gotten very confused.
 
> I removed all the ED Market Connector files and directories that I can locate. Add/Remove is very confused now thinking the program still exists and I am unable to execute the Uninstall.

If `Windows Settings` > `Apps` > search for 'Market' > Click to select and click 'Uninstall' doesn't work then I'll have to do some investigation on how to force an uninstall (all your EDMC settings will remain safe in the registry, as well third-party plugins assuming you put them in the correct location).  You might end up having to manually remove relevant registry keys so that Windows believes it is *not* installed so as to allow installing 5.0.2 again.

In future try the Apps list 'Uninstall' option **first**, not manual removal of files.

One thing to try is to:

1. download a fresh copy of the 5.0.2 MSI (to be sure) from https://github.com/EDCD/EDMarketConnector/releases/download/Release%2F5.0.2/EDMarketConnector_win_5.0.2.msi 
2. Open a 'Command Prompt', i.e. cmd.exe
3. Navigate using `cd <path>` to where you downloaded the file, e.g. `cd C:\Users\YOURUSER\Downloads`
3. Run: `msiexec /f EDMarketConnector_win_5.0.2.msi` - the `/f` forces a repair reinstall.

**Comment by srseibs** [comment]
This did not work:
msiexec /f EDMarketConnector_win_5.0.2.msi
Windows Installer error: "This action is only valid for products that are
currently installed"

[image: image.png]

**Reply by Athanasius** [reply]
And what happens if you try:

`msiexec /f a EDMarketConnector_win_5.0.2.msi` ?

If that fails then try:

`msiexec -L * edmc-install.log EDMarketConnector_win_5.0.2.msi`

and then drag and drop the edmc-install.log file into a comment here.

Or you could read the [msiexec documentation](https://docs.microsoft.com/en-us/windows/win32/msi/command-line-options) yourself and see if any other options help.

**Comment by srseibs** [comment]
*TL;DR*:  Ok - I got it.

Those two msiexec commands provided are syntactically wrong, but I get the
idea. They still will not proceed since the installer thinks there is an
existing installation. My add/remove programs is stuck with version 5.0.0:
[image: image.png]

*The fix for me was to remove this program entry from the registry
using:  **Microsoft
Install and Uninstall Troubleshooter*
<https://support.microsoft.com/en-us/topic/fix-problems-that-block-programs-from-being-installed-or-removed-cca7d1b6-65a9-3d98-426b-e9f927e1eb4d>
 This successfully removed the entry from my programs list.

Then the *msiexec *installation ran as expected. The EDMC popped right up,
in the same location, with the correct data and the last Cmdr without any
interaction from me. The Add/Remove program list properly shows 5.0.2.0

Not sure what happened here. I *never* remove files/dirs from programs like
this (I only did that in this case out of desperation, *AFTER *unsuccessfully
trying the Uninstall). I accept all the EDMC market connector updates
religiously. Clearly something bad with an update after 5.0.0.0 and maybe
the updates have been silently failing. As I mentioned originally, I think
the updates used to automatically restart the Connector, and they did not
recently, so that supports the idea that the update was silently failing.

Anyway - thanks for your help, and even more important, thanks for your
work on this awesome tool. I cannot imagine playing Elite without this. The
combination of EDMC + Inara is an incredible, synergistic awesomeness.

- o7 Mike (Cmdr Dr. Smith)

**Reply by Athanasius** [reply]
Do you have any Anti-Virus (other than Windows Defender which never gives me trouble) at play?  We did have multiple reports of false positives with many different AV and 5.0.0 (the extra .0 is only a Windows thing, long story).

If an AV had decied it didn't initially like something about 5.0.0, but then OK'd it, but didn't completely clean up after itself then it could have left things in a mess.

**Comment by srseibs** [comment]
Norton AV - no threat activity logged during the Odyssey time frame....

**Comment by Athanasius** [comment]
I just added to our Troubleshooting page: https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#help-edmc-isn-39-t-properly-installed-and-i-can-39-t-uninstall-or-upgrade-it

thanks for the heads up about that tool 💯
